meter/rct.go

```diff
@@ -0,0 +1,156 @@
+package meter
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/evcc-io/evcc/api"
+	"github.com/evcc-io/evcc/util"
+
+	"github.com/mlnoga/rct"
+)
+
+/*
+This meter supports devices implementing the RCT communication protocol, e.g. the RCT PS 6.0 with / without battery.
+
+** Usages **
+The following usages are supported:
+- grid    ... for reading the power imported or exported to the grid
+- pv      ... for reading the power produced by the pv
+- battery ... for reading the power imported or exported to the battery
+
+** Example configuration **
+meters:
+- name: GridMeter
+  type: rct
+  host: 192.168.1.23
+  cache: 2s
+  usage: grid
+- name: PvMeter
+  type: rct
+  host: 192.168.1.23
+  cache: 2s
+  usage: pv
+- name: BatteryMeter
+  type: rct
+  host: 192.168.1.23
+  cache: 2s
+  usage: battery
+
+** Limitations **
+It is not allowed to provide different URIs or passwords for different lgess meters since always the
+same hardware instance is accessed with the different usages.
+*/
+
+// RCTMeter implements the api.Meter interface
+type RCTMeter struct {
+	conn    *rct.Connection // connection with the RCT device
+	usage    string         // grid, pv, battery
+}
+
+func init() {
+	registry.Add("rct", NewRCTFromConfig)
+}
+
+//go:generate go run ../cmd/tools/decorate.go -f decorateRCT -b api.Meter -t "api.MeterEnergy,TotalEnergy,func() (float64, error)" -t "api.Battery,SoC,func() (float64, error)"
+
+// NewRCTFromConfig creates an RCTMeter Meter from generic config
+func NewRCTFromConfig(other map[string]interface{}) (api.Meter, error) {
+	cc := struct {
+		Host, Usage, Cache string
+	}{}
+
+	if err := util.DecodeOther(other, &cc); err != nil {
+		return nil, err
+	}
+	cacheDuration,err:=time.ParseDuration(cc.Cache)
```

Comment:
Remove

Reply:
Not quite sure what to do here. Will changing the type of Cache automatically use ParseDuration?

Reply:
I've changed to to  a `time.Duration`- mapstructure will already take care of that